poros/game_results.rb

```diff
@@ -0,0 +1,72 @@
+
+class GameResults
+  
+  def service 
+    GameService.new
+  end
+
+  def game_params_by_name(game_name)
+    params = service.get_games_by_name(game_name)[0]
+    result = {data: {
+      age_ratings:  run_array(:get_age_ratings, :rating, params[:age_ratings]), # array of ids
+      release_date: Time.at(params[:first_release_date]).year, # Unix Time
+      cover:        "https:#{cover_url(params[:cover])}",
+      popularity:   params[:popularity], # good
+      summary:      params[:summary], # good
+      name:         params[:name], # good
+      total_rating: params[:total_rating], # good
+      categories:   params[:category], # got 0
+      genres:       run_array(:get_game_genres, :name, params[:genres]), # array of ids
+      keywords:     params[:keywords], # array of ids -- several hundred :(
+      platforms:    run_array(:get_platforms, :name, params[:platforms]), # array of ids
+      similars:     run_array(:get_games_by_id, :name, params[:similar_games]), # array of ids
+      screenshots:  format_screenshots(params), # array of ids
+      themes:       run_array(:get_game_themes, :name, params[:themes]), # array of ids
+      videos:       "https://www.youtube.com/watch?v=#{youtube_id(params[:videos][0])}" # good -> array of ids
+    }}
+  end
+
+  def run_array(method, key, array)
+    array.map do |element|
+      service.send(method,element)[0][key]
+    end
+  end
+
+  def cover_url(cover_id)
+    service.get_cover(cover_id)[0][:url] # id
+  end
+  
+  def youtube_id(video_id)
```

Comment:
Kind of curious if we even need a youtube api or could we just use the embedded code and pass the youtube video id from the api in here. This currently goes to the kissing booth 2 but we could just pass whatever video code the api returns and pass in here in our app.

```
<iframe width="560" height="315" src="https://www.youtube.com/embed/**fjVonI2oVeM**" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```